sys/sysent.c

```diff
@@ -14,20 +15,29 @@ int sys_nosys(thread_t *td, syscall_args_t *args) {
 /* This is just a stub. A full implementation of this syscall will probably
    deserve a separate file. */
 int sys_write(thread_t *td, syscall_args_t *args) {
+  int retval;
   int fd = args->args[0];
   const char *buf = (const char *)(uintptr_t)args->args[1];
   size_t count = args->args[2];
 
-  kprintf("[syscall] write(%d, %p, %zu)\n", fd, buf, count);
+  log("sys_write(%d, %p, %zu)", fd, buf, count);
 
   /* TODO: copyout string from userspace */
   if (fd == 1 || fd == 2) {
-    kprintf("%.*s", (int)count, buf);
+    char kbuf[80];
+    size_t done = min(count, sizeof(kbuf));
 
-    return count;
+    retval = copyin(buf, kbuf, done);
```

Comment:
Yes, I understand, this **is** correct, but an extra test case confirming that `sys_write` does not fail in such scenario might be useful. Though I might just as well add such test case to #152, so do go ahead and merge this branch.